### PR TITLE
Add datasource whitelist

### PR DESCRIPTION
**Pull request by ewelinajozwik**
Related pull requests:

- Fractal.js: https://github.com/thehyve/Fractal.js/pull/6
- Glowing Bear: https://github.com/thehyve/glowing-bear/pull/627
- Puppet module for Glowing Bear: https://github.com/thehyve/puppet-glowing_bear/pull/12

Related ticket: [TMT-635](https://jira.thehyve.nl/browse/TMT-635)

**Issue summary:**

Glowing Bear used to pass to Fractalis which datasource URL to connect to and which keycloak instance to use for token verification. Client should not be trusted for these values. 

List of all supported sources will now be part of the configuration, together with a mapping to a handler type - in general each handler should use one authorization server. If there are multiple transmart backends, each of them should be connected to the same Keycloak, using the same realm and client.

Example configuration:

```
DATA_SERVICES = {
    'data-services': {
        'transmart-dev': {
            'handler': 'transmart',
            'server': 'https://transmart-dev.thehyve.net'
        },
        'transmart-test': {
            'handler': 'transmart',
            'server': 'https://transmart-test.thehyve.net'
        }
    }
}
```

Example use:
```
POST /data
{
  “service”: “transmart-dev”,
  “descriptors”: {
    “data_type”: “numerical”,
    “label”: “Age”,
    “constraint”: {“type”: “concept”, “conceptCode”: ”Individual.age”}
  }
}
```

### CODE REVIEW

File: fractalis/config.py [hunk 1]
```diff
@@ -43,4 +43,13 @@
 OIDC_SERVER_URL = 'https://keycloak-example.com/auth/realms/transmart-realm/'
 OIDC_CLIENT_ID = 'transmart-client'
 
+DATA_SERVICES = {
+    'data_services': {
+        'test-service': {
```
Comment (ewelinajozwik) [inline fractalis/config.py]: This specific configuration is required for tests to pass. There should be a separate config for functional/unit test purposes, but I did not find a way to properly use another config, when running tests...

Comment (gijskant) [reply fractalis/config.py]: The trick to change the config in the test code does not work? E.g., here: https://github.com/thehyve/Fractalis/blob/dev/tests/functional/test_data.py#L282

Comment (ewelinajozwik) [reply fractalis/config.py]: unfortunately not, data_services_config is apparently assigned before tests start: https://github.com/thehyve/Fractalis/blob/datasource-whitelist/fractalis/__init__.py#L44 and I tried a few ways to override `data_services_config` in tests, but it did not work.

Comment (ewelinajozwik) [reply fractalis/config.py]: fixed by https://github.com/thehyve/Fractalis/blob/datasource-whitelist/tests/__init__.py

Comment (gijskant) [reply fractalis/config.py]: Yes, looks great now

**Review by gijskant — APPROVED** [review]

**Pull request merged**